conans/client/build/layout.py

```diff
@@ -0,0 +1,119 @@
+import os
+
+
+class Layout(object):
+    def __init__(self, conan_file):
+        self._conan_file = conan_file
+
+        # Build layout
+        self.build = "build"
+        self.src = ""
+        # TODO: To discuss, are we sure we want only 1 value?
+        self.build_libdir = ""  # Relative to self.build
+        # TODO: To discuss, are we sure we want only 1 value?
+        self.build_bindir = ""  # Relative to self.build
+        self.build_includedirs = [""]  # Relative to self.build (and or self.src)
+        self.build_installdir = None  # Not relative to self.build
+
+        self.pkg_libdir = "lib"
+        self.pkg_bindir = "bin"
+        self.pkg_includedir = "include"
+        self.pkg_builddir = "build"
+
+    def __str__(self):
+        return str(self.__dict__)
+
+    # Getters, relative to base
+    @property
+    def build_lib_folder(self):
+        return os.path.join(self.build, self.build_libdir)
+
+    @property
+    def build_bin_folder(self):
+        return os.path.join(self.build, self.build_bindir)
+
+    @property
+    def build_include_folders(self):
+        return [os.path.join(self.build, f) for f in self.build_includedirs]
+
+    @property
+    def build_install_folder(self):
+        return self.build_installdir if self.build_installdir is not None else self.build
+
+    # Getters, useful for recipes, relative to self._conanfile.xxx
+    @property
+    def build_folder(self):
+        return os.path.join(self._conan_file.build_folder, self.build)
+
+    @property
+    def source_folder(self):
+        return os.path.join(self._conan_file.source_folder, self.src)
+
+    def package(self):
+        # FIXME: Not very flexible. Useless? What if I want to copy only some patterns
+        #        Good enough for default?
+        #        Proposal kwargs: header_patterns=None, lib_patterns=None, exe_patterns=None
+        #        2Proposal: Declare the patterns in the layout too?
+        for include in self.build_includedirs:
```

Comment:
Yes, I think this is still wrong. The includedirs, unless we add 2 variables instead of 1, have to be relative to `self.build` or `self.src`. Here is missing prepending the `self.build`/`self.src` to the include value. This is still not ideal, the real behavior should be to use `self.build` when the copy acts over `conanfile.build_folder` and `self.src` when the copy acts over `conanfile.source_folder` (if different).

Reply:
I changed a couple of things. I think the easier is that `build_includedirs` are not relative to `lyt.build` nor `lyt.src` but defaulted to `[lyt.build, lyt.src]` the rest of the code is valid. I'll add more test to see that the layout is taken into account both packaging and in the build info of consumers (specially editables)